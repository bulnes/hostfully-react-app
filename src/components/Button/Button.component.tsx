interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  buttonType?: "primary" | "secondary" | "danger";
  children: React.ReactNode;
}

export function Button({ children, buttonType, ...attributes }: ButtonProps) {
  let buttonClasses;

  switch (buttonType) {
    case "secondary":
      buttonClasses = "bg-white text-blue-700 border border-blue-500";
      break;
    case "danger":
      buttonClasses = "bg-white text-red-700 border border-red-500";
      break;
    case "primary":
    default:
      buttonClasses =
        "bg-blue-500 hover:bg-blue-700 text-white border border-blue-500";
  }

  return (
    <button
      className={`${buttonClasses} font-bold py-2 px-4 rounded`}
      {...attributes}
    >
      {children}
    </button>
  );
}
