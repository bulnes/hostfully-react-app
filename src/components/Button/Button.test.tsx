import { render } from "@testing-library/react";
import { Button } from "./Button.component";

describe("Component: Button", () => {
  it("should be defined", () => {
    expect(Button).toBeDefined();
  });

  it("should render correctly", () => {
    const wrapper = render(<Button>Click me!</Button>);
    expect(wrapper).toBeDefined();
  });

  it("should render children correctly", () => {
    const wrapper = render(<Button>Click me!</Button>);
    expect(wrapper.getByText("Click me!")).toBeDefined();
  });

  it("should render primary button correctly", () => {
    const wrapper = render(<Button buttonType="primary">Click me!</Button>);
    expect(wrapper.getByText("Click me!")).toMatchSnapshot();
  });

  it("should render secondary button correctly", () => {
    const wrapper = render(<Button buttonType="secondary">Click me!</Button>);
    expect(wrapper.getByText("Click me!")).toMatchSnapshot();
  });

  it("should render danger button correctly", () => {
    const wrapper = render(<Button buttonType="danger">Click me!</Button>);
    expect(wrapper.getByText("Click me!")).toMatchSnapshot();
  });
});
