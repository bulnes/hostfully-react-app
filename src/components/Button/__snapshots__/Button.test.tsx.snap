// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Button should render danger button correctly 1`] = `
<button
  class="bg-white text-red-700 border border-red-500 font-bold py-2 px-4 rounded"
>
  Click me!
</button>
`;

exports[`Component: Button should render primary button correctly 1`] = `
<button
  class="bg-blue-500 hover:bg-blue-700 text-white border border-blue-500 font-bold py-2 px-4 rounded"
>
  Click me!
</button>
`;

exports[`Component: Button should render secondary button correctly 1`] = `
<button
  class="bg-white text-blue-700 border border-blue-500 font-bold py-2 px-4 rounded"
>
  Click me!
</button>
`;
